src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ObjectCreationCompletionProviderTests.cs

```diff
@@ -769,5 +769,34 @@ class Program
 }";
             await VerifyProviderCommitAsync(markup, "List<int>", expectedMark, commitChar: ';');
         }
+
+        [Theory]
+        [WorkItem(51629, "https://github.com/dotnet/roslyn/issues/51629")]
+        [InlineData('.')]
+        [InlineData(';')]
+        public async Task DoNotAddParenthesisWhenContructorIsInaccessible(char commitChar)
+        {
+            var source = @"
+Outer x = new Out$$
+class Outer
+{
+    private Outer() { }
+
+    public class Inner
+    {
+    }
+}";
+            var expected = $@"
+Outer x = new Outer{commitChar}
```

Comment:
I think user's intention here is very clear, they wants `Outer x = new Outer()` (as we inferred the object type and they have typed `new` already), so we might as well be consistent and let them do as they intended regardless of the accessibility of the ctor given that this will ended up with the same error either way.

Reply:
> I think user's intention here is very clear, they wants `Outer x = new Outer()`

In that case I think the `commitChar` is going to be `;`, in this case we should always add `()` regardless of nested types or constructor accessibility checks.

The main problem is only when `commitChar` is `.`

Reply:
Yes it is definitely possible and we can add checks for such, but I suspect it'd be so rare (I don't mean the type declaration pattern, but how objects of the nested types are created) therefore not a pattern we care to make a special case to accommodate tbh. So for ObjectCreationProvider, I'd be totally fine with always adding "()"